src/components/KSelect/KSelect.vue

```diff
@@ -555,63 +418,43 @@ const popoverContentMaxHeight = computed((): string => getSizeFromString(props.d
 // TypeScript complains if I bind the original object
 const boundKPopAttributes = computed(() => ({ ...createKPopAttributes.value }))
 
-const filteredItems = computed(() => {
-  // For autosuggest, items don't need to be filtered internally
-  return props.autosuggest ? selectItems.value : props.filterFunc({ items: selectItems.value, query: filterStr.value })
-})
-
-const placeholderText = computed((): string => {
-  if (props.placeholder) {
-    return props.placeholder
-  } else if (attrs.placeholder) {
-    return attrs.placeholder as string
-  }
-  if (props.appearance === 'button' || !filterIsEnabled.value) {
-    return 'Select an item'
-  }
-  return 'Filter...'
-})
-
-const selectButtonText = computed((): string => {
-  if (props.buttonText && selectedItem.value) {
-    return props.buttonText
-  } else if (selectedItem.value) {
-    return selectedItem.value.label
-  }
-  return placeholderText.value
-})
+const placeholderText = computed((): string => props.placeholder || attrs.placeholder as string || 'Filter...')
```

Comment:
Should the default placeholder be `Select` instead? If filtering isn't enabled it doesn't make much sense